contracts/starknet/voting_strategies/whitelist.cairo

```diff
@@ -0,0 +1,48 @@
+%lang starknet
+
+from starkware.cairo.common.uint256 import Uint256
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from contracts.starknet.lib.eth_address import EthAddress
+
+@storage_var
+func whitelist(address : EthAddress) -> (voting_power : Uint256):
+end
+
+@event
+func whitelisted(address : EthAddress):
+end
+
+func register_whitelist{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr : felt}(
+        _whitelist_len : felt, _whitelist : felt*):
+    if _whitelist_len == 0:
+        return ()
+    else:
+        let address = EthAddress(_whitelist[0])
+        # Add it to the whitelist
+        let vp = Uint256(_whitelist[1], _whitelist[2])
+        whitelist.write(address, vp)
+
+        # Emit event
+        whitelisted.emit(address)
```

Comment:
shall we emit the voting power of the address here too?

Reply:
good idea! 7a3f8d871276878a1f1b32e907b9d5b3d71cd9ed :D